### ISSUE TITLE
[BUG] gaussian likelihood computation 

### ISSUE THREAD

**Issue by tominku** [takuseno/d3rlpy#192]
======== dynamics.py ===========

def _gaussian_likelihood(
    x: torch.Tensor, mu: torch.Tensor, logstd: torch.Tensor
) -> torch.Tensor:
    inv_std = torch.exp(-logstd)
    return (((mu - x) ** 2) * inv_std).mean(dim=1, keepdim=True)

======= I think It should be... =============

def _gaussian_likelihood(
    x: torch.Tensor, mu: torch.Tensor, logstd: torch.Tensor
) -> torch.Tensor:
    inv_std = torch.exp(-logstd)
    return 0.5 * (((mu - x) ** 2) * (inv_std ** 2)).sum(dim=1, keepdim=True)

![image](https://user-images.githubusercontent.com/25144781/172034551-ca2629d6-f733-44dc-96cd-035319fbf9c5.png)

**Comment by takuseno** [comment]
@tominku Thanks for the issue. I didn't follow the math yet, but, the current implementation follows the official repository (more specifically, the official MBPO repository, which MOPO is based on).
https://github.com/jannerm/mbpo/blob/ac694ff9f1ebb789cc5b3f164d9d67f93ed8f129/mbpo/models/bnn.py#L582

**Comment by takuseno** [comment]
Ah, wait, probably, you're right. it looks like it should be:
```
return (((mu - x) ** 2) * (inv_std ** 2)).mean(dim=1, keepdim=True)
```
I'll fix this later.

**Comment by takuseno** [comment]
It's fixed in this commit https://github.com/takuseno/d3rlpy/commit/384e28365a07f34ac9162a8a2af18349ba93a0a0 .

**Comment by takuseno** [comment]
Let me close this issue since it's resovled.

**Comment by tominku** [comment]
In addition to that, I think the mean() part should be sum(), because the math formula says that.

**Comment by takuseno** [comment]
I don't think so. The multi-dimensional gaussian distribution is usually treated like this in RL. Please check the official repository too.

**Comment by tominku** [comment]
They (the link you provided) used mean() for both var_loss and mse_loss consistently,
but your implementation uses sum() for var_loss, and mean() for mse_loss, which is inconsistent.
I want you to look at the math formula and compare with the code. 

loss = likelihood_loss + penalty + 1e-2 * bound_loss

The penalty part use sum() (over dim=1), but likelihood_loss part use mean() (over dim=1), which is inconsistent. I want you to think about it again.

**Comment by tominku** [comment]
I also hope this pytorch-based model-based rl library helps.

https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/models/gaussian_mlp.py
https://github.com/facebookresearch/mbrl-lib

**Comment by takuseno** [comment]
I see. Could you PR the change instead? I feel that should be faster.

**Comment by tominku** [comment]
![image](https://user-images.githubusercontent.com/25144781/172369427-7faecf0c-8ddf-4df3-91ce-45951e058a2e.png)

**Comment by tominku** [comment]
since we are dealing with the diagonal covariance matrix, the inverse of the covariance can be easily computed, as the code says.

**Comment by takuseno** [comment]
It's easier to check what's changed if you make a PR since it provides cleaner diffs.

**Comment by takuseno** [comment]
Let me close this issue since we've merged PR #194 . Thank you for your contribution!
